samples/BoltOn.Samples.Application/Handlers/UpdateStudentHandler.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using BoltOn.Data;
+using BoltOn.Logging;
+using BoltOn.Mediator.Pipeline;
+using BoltOn.Samples.Application.Entities;
+
+namespace BoltOn.Samples.Application.Handlers
+{
+	public class UpdateStudentRequest : IRequest
+	{
+		public Guid StudentId { get; set; }
+
+		public string FirstName { get; set; }
+
+		public string LastName { get; set; }
+
+		public int StudentTypeId { get; set; }
+	}
+
+	public class UpdateStudentHandler : IRequestAsyncHandler<UpdateStudentRequest>
+	{
+		private readonly IRepository<Student> _studentRepository;
+		private readonly IBoltOnLogger<CreateStudentHandler> _logger;
+		private readonly IRepository<StudentType> _studentTypeRepository;
+
+		public UpdateStudentHandler(IRepository<Student> studentRepository,
+			IBoltOnLogger<CreateStudentHandler> logger,
+			IRepository<StudentType> studentTypeRepository)
+		{
+			_studentRepository = studentRepository;
+			_logger = logger;
+			_studentTypeRepository = studentTypeRepository;
+		}
+
+		public async Task HandleAsync(UpdateStudentRequest request, CancellationToken cancellationToken)
+		{
+			_logger.Debug("Updating student...");
+			var studentType = await _studentTypeRepository.GetByIdAsync(request.StudentTypeId);
+			var student = await _studentRepository.GetByIdAsync(request.StudentId, cancellationToken);
+			student.Update(request, studentType.Description);
+			await _studentRepository.UpdateAsync(student, cancellationToken);
```

Comment:
@gokulm : instead of having two updates one using student object and one for actual update of repo , as student repo is anyways cqrsrepository can we override updateAsync method in the cqrsrepository which does the raise event first and then call base update of repository method ? so user doesn't have to worry abt doing both?

Reply:
the next version will not have a separate cqrsrepository, there will be only one repository (checkout cosmos branch). also, the domain entity's method will not be Update, as this is just a sample, i named it Update, but it will be a method that conveys some business action